src/subhalo_tracking.cpp

```diff
@@ -529,18 +758,63 @@ void SubhaloSnapshot_t::FeedCentrals(HaloSnapshot_t &halo_snap)
     else
     {
       auto &central = Subhalos[Members[0]];
-      assert(central.Particles.size());
-      central.Particles.swap(Host.Particles); // reuse the halo particles
+
+      // Only test whether we have particles for previously-resolved subhaloes
+      if (central.IsAlive())
+        assert(central.Particles.size());
+
+      /* The subhalo now contains all the host particles. Those belonging to
+       * subhaloes will be masked during unbinding, if exclusive mass option is
+       * used. */
+      central.Particles.swap(Host.Particles);
       central.Nbound = central.Particles.size();
+      bool tracerIndexSet = false;
+
+      // Use the most bound tracer from the previous snapshot that remains
+      // in the FOF group as the tracer. We need this information for the masking.
+
+      /* This is for resolved subhaloes. Automatically skipped if we are doing
+       * an orphan */
+      for (HBTInt i = 0; i < Host.Particles.size(); i++)
```

Comment:
Do we still need to do this? I believe it was added to assign a tracer to potentially new orphans, since they explicitly required a particle. Now we rely on the ID that we set as soon as we load the particle information, and is hence decoupled from the presence of a particle or not.

Reply:
I agree that this is no longer needed, it was implemented just to prevent tracerless orphans. If this gets removed we would need to test to make sure we still aren't getting any problematic orphans.

Reply:
Removed.